sound/soc/sof/ipc4-pcm.c

```diff
@@ -836,5 +836,6 @@ const struct sof_ipc_pcm_ops ipc4_pcm_ops = {
 	.pcm_setup = sof_ipc4_pcm_setup,
 	.pcm_free = sof_ipc4_pcm_free,
 	.delay = sof_ipc4_pcm_delay,
-	.ipc_first_on_start = true
+	.ipc_first_on_start = true,
+	.delayed_platform_trigger = true,
```

Comment:
I find the commit message confusing:

In the case of IPC4, a pipeline is only paused during STOP/PAUSE/RELEASE
triggers and the FW keeps the host DMA running when a pipeline is
paused.

that's the start, so far so good.

With start/stop tests that do not involve a hw_free, the
pipeline switches between pause and release triggers.

You lost me already here. We're mixing start/stop and pause/release in the same sentence. I don't have enough understanding of what the tests do to figure out what you are trying to say.

 Therefore, the
host DMA must not be stopped and restarted during these tests.

which tests? Stop or pause/release? Both? There are too many unclear parts in the description.

Introduce a new flag in struct sof_ipc_pcm_ops, delayed_platform_trigger,
that will be used to ensure that the host DMA will not be stopped during
the STOP/PAUSE/RELEASE triggers.

I think you should clarify what states the DMA can have, I think it's pause/stop/reset, but it's not fully clear to me. @ujfalusi had a similar comment.

Reply:
updated the commit message now @plbossart . Let me know what you think of it now